sncosmo/models.py

```diff
@@ -955,6 +956,70 @@ def _flux(self, phase, wave):
                 self._3d_model_flux(points).reshape(lp, lw))
 
 
+class SNEMOSource(Source):
+    """The SNEMO Type Ia supernova spectral timeseries model
+
+    The spectral flux density of this model is given by
+
+    .. math::
+       F(t, \lambda) = c_0(e_0(t, \lambda) + 
+                           \Sum_{i=1}^{n} c_i e_i(t, \lambda))
+                           \\times FM07(\lambda, A_s)
+    where ``c_0``, ``c_i``, and ``A_s`` are the free parameters of the model.
+
+    Parameters
+    ----------
+    fluxfile : str or obj, optional
+        Filename of an ascii file containing 2-d
+        array of spectral flux density values for a grid of phase
+        and wavelength values. Assuming columns ``phase``, ``wavelength``,
+        ``e_0``, ``e_1``, ``e_2``...
+    n_vector : int
+        One of 2, 7, 15 to select SNEMO2, SNEMO7, SNEMO15
+    """
+    
+    def __init__(self, fluxfile='SNEMO7_ev.dat', name=None,
+                 version=7):
+
+        n_vector = int(float(version))
```

Comment:
It seems like it would be better to determine `n_vector` from the shape of the `values` array after parsing the `fluxfile`. This would ensure consistency between the file contents and `n_vector`. 

In sncosmo, usually the `version` argument to the model matches the `version` in the registry (`1.0` in this case) and doesn't actually affect model behavior - it's just for human reading so you know which model you loaded.

Reply:
Yes, that is more logical. I went back and forth on how to support the three models initially, but an explicit key is not necessary.